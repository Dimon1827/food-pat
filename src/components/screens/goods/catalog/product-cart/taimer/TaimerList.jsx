import Taimer from './Taimer';

function TaimerList({ time }) {
  return (
    <ul className="additional-products__list">
      <Taimer time={time} />
    </ul>
  );
}

export default TaimerList;
