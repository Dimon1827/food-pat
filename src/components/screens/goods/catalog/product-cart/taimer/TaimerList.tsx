import Taimer from './Taimer';
import { ITaimer } from './taimer.types';

function TaimerList({ time }: ITaimer) {
  return (
    <ul className="additional-products__list">
      <Taimer time={time} />
    </ul>
  );
}

export default TaimerList;
