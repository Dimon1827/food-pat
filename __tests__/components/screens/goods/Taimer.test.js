import { render, screen } from '@testing-library/react';
import TaimerList from '@/components/screens/goods/catalog/product-cart/taimer/TaimerList';

const time = {
  days: 3,
  hours: 11,
  minutes: 18,
};

describe('TaimerList', () => {
  it('TaimerList renders correctly', () => {
    render(<TaimerList time={time} />);

    expect(screen.getByRole('list')).toMatchSnapshot();
  });
  it('TaimerList should render null if time is null or undefined', () => {
    render(<TaimerList/>);

    const timeElement = screen.queryByText(/./, {
      selector: '.additional-products__number',
    }); 
    expect(timeElement).toBeNull();
  });
});
