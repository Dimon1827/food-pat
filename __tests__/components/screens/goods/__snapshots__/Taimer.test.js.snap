// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaimerList TaimerList renders correctly 1`] = `
<ul
  class="additional-products__list"
>
  <li
    class="additional-products__item"
  >
    <span
      class="additional-products__number"
    >
      3
    </span>
    <p
      class="additional-products__text"
    >
      дня
    </p>
  </li>
  <li
    class="additional-products__item"
  >
    <span
      class="additional-products__number"
    >
      11
    </span>
    <p
      class="additional-products__text"
    >
      часов
    </p>
  </li>
  <li
    class="additional-products__item"
  >
    <span
      class="additional-products__number"
    >
      18
    </span>
    <p
      class="additional-products__text"
    >
      минут
    </p>
  </li>
</ul>
`;
